index.js

```diff
@@ -37,34 +69,31 @@ class Main {
     });
   }
 
-  async disconnect() {
-    // This call should be refactored to work with async/await
-    this.exporter.disconnect();
-    await this.microServer.close();
+  _updateMetrics() {
+    metrics.currentBlock.set(this.worker.lastConfirmedBlock);
+    metrics.requestsCounter.inc(this.worker.getNewRequestsCount());
+    metrics.requestsResponseTime.observe(new Date() - this.worker.lastRequestStartTime);
+    metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
   }
 
   async workLoop() {
     while (this.shouldWork) {
-      const lastRequestStartTime = new Date();
-      const events = await this.worker.work();
-      this.worker.lastExportTime = Date.now();
-      metrics.currentBlock.set(this.worker.lastConfirmedBlock);
+      this.worker.lastRequestStartTime = new Date();
+      const events = await this.worker.work()
+        .catch((err) => { throw new Error(`Error in workloop: ${err.toString()}`); });
```

Comment:
we tend to provide additional info in err messages. this might be helpful. you can add  block number here

Reply:
I'd prefer if we gave that information (which we sometimes do) in the corresponding workers and not in the index file. In this case I'd have to do some gymnastics like
```js
throw new Error(`<error> : ${this.worker.lastExportedBlock}`);
```
which I don't really like. I'd make another PR for this, but the current PR is concentrated on the main file and its functionality